adapters/handlers/grpc/v1/mapping.go

```diff
@@ -192,23 +182,222 @@ func NewObject[P schema.PropertyInterface](v map[string]interface{}, parent P, s
 	return x, nil
 }
 
-// NewList constructs a ListValue from a general-purpose Go slice.
-// The slice elements are converted using NewValue.
-func NewPrimitiveList[T bool | float64 | string](v []T, dt schema.DataType) (*pb.ListValue, error) {
-	var err error
-	x := &pb.ListValue{Values: make([]*pb.Value, len(v))}
-	for i, v := range v {
-		x.Values[i], err = NewPrimitiveValue(v, dt)
+func parseArray[T float64 | bool | string](v interface{}, dt schema.DataType) ([]T, error) {
+	val, ok := v.([]T)
+	if !ok {
+		return nil, protoimpl.X.NewError("invalid type: %T when serializing %v", v, dt.String())
+	}
+	return val, nil
+}
+
+func (m *Mapper) newListValueBool(v interface{}) (*pb.Value, error) {
+	var listValue *pb.ListValue
+	if _, ok := v.([]interface{}); ok {
```

Comment:
Is this present to check for empty lists? 

If yes, could we merge it for the different functions? (After looking at it for a few minutes, I am not sure if it is worth it 🤔 )

Reply:
Because of the specificity of `&pb.ListValue{Kind: &pb.ListValue_BoolValues{BoolValues: &pb.BoolValues{Values: []bool{}}}}` to `bool`, it's better to duplicate it in each method. If we were just doing `&pb.ListValue{Values: []*pb.Value{}}` then I'd agree but the new `ListValue` impl demands a very specific return for each type